src/app/clusters/ias-zone-client/ias-zone-client.cpp

```diff
@@ -390,7 +390,8 @@ static void removeServer(uint8_t * ieeeAddress)
 static EmberStatus sendCommand(EmberNodeId destAddress)
 {
     emberAfSetCommandEndpoints(myEndpoint, emberAfIasZoneClientKnownServers[currentIndex].endpoint);
-    EmberStatus status = emberAfSendCommandUnicast(EMBER_OUTGOING_DIRECT, destAddress);
+    MessageSendDestination destination = MessageSendDestination(MessageSendDestination::OutGoingDirect(destAddress));
```

Comment:
The readability here is not great....  And its not great for the Is() method either, because the template arg is so verbose.

I would almost be tempted to suggest the following:

1. Give the Outgoing* some other names that are maybe not as nice, but don't matter too much because the types are not public API.
2. Have a static `MessageSendDestination::OutgoingDirect` method that takes nodeid and returns `MessageSendDestination`.  It can call the right typed constructor.
3. Have IsOutgoingDirect methods and the like that call the templated Is() with the right type.

It's a bit more hand-written code in the class, but much nicer for consumers....